controllers/front/return.php

```diff
@@ -98,6 +100,13 @@ public function init()
      */
     public function initContent()
     {
+        $customerId = Tools::getValue('customerId');
+        if (Tools::getValue('customerId')) {
+            $customer = new Customer($customerId);
+            if ($customer->secure_key === Tools::getValue('key')) {
+                $this->context = ContextUtility::setCustomerToContext($this->context, $customer);
+            }
+        }
```

Comment:
`CustomerFactory->reacreateFromRequest()` something would this could wrap this logic.